paper/example/ex_mnitst.jl

```diff
@@ -0,0 +1,345 @@
+# This is a simple example of training KnetNLPModels with Lenet5 and MNIST on CPU/GPU
+# For more tools and indetailed example check KnetNLPModelsProblems.jl
+# Note that we need to fine-tune the example so it has a better results than SGD.
+
+using CUDA
+# using IterTools
+using JSOSolvers
+using LinearAlgebra
+using Random
+using Printf
+using NLPModels
+using SolverCore
+using Plots
+using Knet, Images, MLDatasets
+using Statistics: mean
+using KnetNLPModels
+using Knet:
+    Knet,
+    conv4,
+    pool,
+    mat,
+    nll,
+    accuracy,
+    progress,
+    sgd,
+    param,
+    param0,
+    dropout,
+    relu,
+    minibatch,
+    Data
+
+# ####################################################
+# # used in the callback of R2 for training deep learning model 
+# ####################################################
+
+mutable struct StochasticR2Data
+    epoch::Int
+    i::Int
+    # other fields as needed...
+    max_epoch::Int
+    acc_arr::Vector{Float64}
+    train_acc_arr::Vector{Float64}
+    epoch_arr::Vector{Float64}
+    grads_arr::Vector{Float64} #TODO fix this type to be dynamic
+    ϵ::Float64 #TODO Fix with type T
+end
```

Comment:
Is this not defined in JSOSolver alongside R2Solver?

Reply:
Not really,
This is what we need to keep track of and we specify it for this example 
R2 doesn't have this